From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeff Genovy <29107334+jefgen@users.noreply.github.com>
Date: Fri, 20 Nov 2020 12:30:15 -0800
Subject: MSFT-PATCH: Don't use the extended ICU data for Windows OS ICU.

For the Windows OS build of ICU, we only have one data file
and we don't use the extended data at all. We make this function a no-op
in order to save a few cycles for perf, but more importantly so that
we don't try to load a versioned data file (ex: icudt68l.dat) after
already loading the non-versioned common data file.

diff --git a/icu/icu4c/source/common/udata.cpp b/icu/icu4c/source/common/udata.cpp
index ec9c999cea41a6e4c2ab27cf2e6d07e35c43b189..37f7da401c4f8feae4d5380e7903b06e0ef2779d 100644
--- a/icu/icu4c/source/common/udata.cpp
+++ b/icu/icu4c/source/common/udata.cpp
@@ -820,6 +820,16 @@ openCommonData(const char *path,          /*  Path from OpenChoice?          */
  *----------------------------------------------------------------------*/
 static UBool extendICUData(UErrorCode *pErr)
 {
+// MSFT-Change: For the Windows OS build of ICU, we only have one data file
+// and we don't use the extended data at all. We make this function a no-op
+// in order to save a few cycles for perf, but more importantly so that
+// we don't try to load a versioned data file (ex: icudt68l.dat) after
+// already loading the non-versioned common data file.
+#if defined(ICU_DATA_DIR_WINDOWS)
+    (void)pErr; // suppress unused variable.
+    return FALSE;
+#endif
+
     UDataMemory   *pData;
     UDataMemory   copyPData;
     UBool         didUpdate = FALSE;
